Framework/Geometry/src/Instrument/StructuredDetector.cpp

```diff
@@ -320,6 +325,9 @@ void StructuredDetector::initialize(size_t xPixels, size_t yPixels,
   if (x.size() != (size_t)((m_xPixels + 1) * (m_yPixels + 1)))
     throw std::invalid_argument("StructuredDetector::initialize(): x.size() "
                                 "should be = (xPixels+1)*(yPixels+1)");
+  if (!(beamDirection.compare("z") == 0))
```

Comment:
This is good enough as a check. Some extra comments explaining that this is particular to the StructuredDetector type would help users receiving this error. Some code comments would also be useful.